lib/buffer.js

```diff
@@ -657,7 +658,7 @@ const encodingOps = {
     encoding: 'ascii',
     encodingVal: encodingsMap.ascii,
     byteLength: (string) => string.length,
-    write: (buf, string, offset, len) => buf.asciiWrite(string, offset, len),
+    write: (buf, string, offset, len) => console.error("###", typeof buf, typeof string, typeof offset, typeof len) || stringWrite(buf, string, offset, len),
```

Comment:
You're using `console` in the function. It's just a hunch but I wouldn't be surprised if that prevented V8 from optimizing it.

Reply:
Doesn't make any difference. It never calls `FastWriteString`.